### PR TITLE
CLOUDP-59015: Upload rpm to mongo yum repo

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ [CLOUDP-59015](https://jira.mongodb.org/browse/CLOUDP-59015)

## Further comments
Running test here https://evergreen.mongodb.com/version/5e84863cd1fe0775bdea7566

### CODE REVIEW

File: .evergreen.yml [hunk 1]
```diff
@@ -122,6 +122,50 @@ tasks:
           permissions: public-read
           content_type: ${content_type|application/x-gzip}
 
+  - name: publish
```
Comment (tychoish) [inline .evergreen.yml]: call this push.

File: .repo_config.yml [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+        <tr><td>
+          <address>{{ .RepoName }}</address>
+        </td></tr>
+      </table>
+     </body>
+    </html>
+
+
+repos:
```
Comment (tychoish) [inline .repo_config.yml]: I'd trim the list of repos to be only things you intend to actual push.

File: .evergreen.yml [hunk 3]
```diff
@@ -122,6 +122,50 @@ tasks:
...
+            ./curator repo submit \
+              --service ${barque_url} \
+              --config .repo_config.yml \
+              --distro rhel70 \
+              --edition org \
+              --version $VERSION \
+              --arch x86_64 \
+              --packages https://s3.amazonaws.com/mongodb-mongocli-build/${project}/${revision}/dist/mongocli_next_${revision}_${version_id}_${created_at}_linux_x86_64.rpm
+
+            ./curator repo submit \
```
Comment (tychoish) [inline .evergreen.yml]: I'd do what you can to try and get org/community to run in parallel.

Comment (gssbzn) [reply .evergreen.yml]: yes, I was just thinking of this seeing how long this takes, I may need to run other distros in parallel as well

Comment (tychoish) [reply .evergreen.yml]: there's basically no local work, it's all busy-waiting, so finding some way to not serialize these tasks is good (and largely, all the work is parallelized on the backend.

**Comment by gssbzn** [comment]
@tychoish this is RFAL, opened a build ticket as it looks like we need some work for other rpm distros that are not rehad7

**Comment by tychoish** [comment]
what is RFAL?

File: .evergreen.yml [hunk 4]
```diff
@@ -122,11 +122,106 @@ tasks:
...
-  - name: publish
-    display_name: Building & Publishing
+  - name: package
+    display_name: Packaging
     run_on:
       - rhel70-small
     tasks:
       - name: compile
       - name: package
+  # Redhat buildvariants
```
Comment (tychoish) [inline .evergreen.yml:170]: I'd just append a bunch of tasks to a single build variant.

**Comment by tychoish** [comment]
is the thing in the package only a go binary? does it link to any system library. If contents of the package are the same regardless, you should be able to push the same .rpm all RPM repositories

**Comment by gssbzn** [comment]
@tychoish changed this to be parallel tasks instead of buildvariants,
Yes, the RPM is just a go binary, no dependencies or linking other binaries, we still get 
```
[2020/04/02 11:55:45.836] [curator] 2020/04/02 11:55:45 [p=emergency]: job 'build-repo.distro.rpm.repo.13b13efe-ee5e-4f91-9868-ec7ea97b5519' completed with error [problem building repo in '/data/repobuilder/repo.mongodb.com/yum/redhat/7/mongodb-enterprise/development/x86_64': signing release metadata for /data/repobuilder/repo.mongodb.com/yum/redhat/7/mongodb-enterprise/development/x86_64: problem with notary service client signing file: exit status 1]
```
Or
```
[2020/04/02 11:43:09.697] [curator] 2020/04/02 11:43:09 [p=emergency]: job 'build-repo.distro.rpm.repo.b83164ac-440c-4769-a513-dabab9e04d4c' completed with error [problem building repo in '/data/repobuilder/repo.mongodb.com/yum/amazon/2/mongodb-enterprise/development/x86_64': signing release metadata for /data/repobuilder/repo.mongodb.com/yum/amazon/2/mongodb-enterprise/development/x86_64: problem with notary service client signing file: exit status 1]
```
[2020/04/02 11:43:09.697] [curator] 2020/04/02 11:43:09 [p=emergency]: job 'build-repo.distro.rpm.repo.b83164ac-440c-4769-a513-dabab9e04d4c' completed with error [problem building repo in '/data/repobuilder/repo.mongodb.com/yum/amazon/2/mongodb-enterprise/development/x86_64': signing release metadata for /data/repobuilder/repo.mongodb.com/yum/amazon/2/mongodb-enterprise/development/x86_64: problem with notary service client signing file: exit status 1]
```

And no clear idea or reference if this is just miss-configuration on our side

**Comment by gssbzn** [comment]
@tychoish any other comment here? I also updated BUILD-10839 with more details

**Pull request merged**